### PR TITLE
Api fastify

### PR DESCRIPTION
**Pull request by deluxor**
I took the initiative to make a major refactor to the project in order to facilitate infrastructure developments and support.
Please check the README.md

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [12.x]
+        node-version: [13.x]
```
Comment (hugoduraes) [inline .github/workflows/nodejs.yml]: I think we should stick with an LTS version of Node.

File: db/models/network.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		},
+		timestamp: {
+			type: DataTypes.DATE,
+			allowNull: true,
+			defaultValue: 'timezone(utc'
+		},
+		unix_ts: {
+			type: DataTypes.INTEGER,
+			allowNull: true,
+			defaultValue: 'date_part(epoch'
```
Comment (hugoduraes) [inline db/models/network.js]: You missed the ending parenthesis

File: db/models/network.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+			allowNull: true,
+			references: {
+				model: 'users',
+				key: 'id'
+			}
+		},
+		timestamp: {
+			type: DataTypes.DATE,
+			allowNull: true,
+			defaultValue: 'timezone(utc'
```
Comment (hugoduraes) [inline db/models/network.js]: You missed the ending parenthesis

File: db/models/users.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+			allowNull: true
+		},
+		info: {
+			type: DataTypes.STRING,
+			allowNull: true
+		},
+		timestamp: {
+			type: DataTypes.DATE,
+			allowNull: true,
+			defaultValue: 'timezone(utc'
```
Comment (hugoduraes) [inline db/models/users.js]: You missed the ending parenthesis

File: db/models/users.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+		},
+		unix_ts: {
+			type: DataTypes.INTEGER,
+			allowNull: true,
+			defaultValue: 'date_part(epoch'
```
Comment (hugoduraes) [inline db/models/users.js]: You missed the ending parenthesis

File: db/schema.sql [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+--Possible Status For A Given User:
+CREATE TABLE IF NOT EXISTS user_status (
+    id serial,
+    status varchar(30)
+);
+CREATE UNIQUE INDEX IF NOT EXISTS user_status_id0 ON user_status (id);
+DELETE FROM user_status;
+INSERT INTO user_status(status) values('normal');
+INSERT INTO user_status(status) values('self quarentine');
```
Comment (hugoduraes) [inline db/schema.sql]: typo, it should be "quarantine"

File: db/schema.sql [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+INSERT INTO user_status(status) values('quarentine');
```
Comment (hugoduraes) [inline db/schema.sql]: same as above

File: package.json [hunk 8]
```diff
@@ -1,38 +1,47 @@
...
-  "description": "",
-  "main": "server.js",
+  "description": "Trackovid-19 API",
+  "main": "app.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node server.js",
-    "start:dev": "nodemon server.js"
+    "start": "node app.js",
+    "dev": "nodemon -w debug -P app.js",
```
Comment (hugoduraes) [inline package.json]: Nodemon is missing on the dev dependencies

File: controllers/v1/case/index.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+module.exports = async (fastify, opts) => {
+  const Case = fastify.sequelize.import('../../../db/models/history.js');
+
+  fastify.post('/case', { schema: { body: fastify.schemas().createCase } }, async (request, reply) => {
+    try {
+      const { postalCode, geo, condition, timestamp, symptoms } = request.body
```
Comment (hugoduraes) [inline controllers/v1/case/index.js]: Is the request.body automatically validated by specifying the schema?

Comment (deluxor) [reply controllers/v1/case/index.js]: Yep, it is.

File: package.json (hunk 8 above)
Comment (deluxor) [reply package.json]: Sure why not?

File: db/models/users.js (hunk 5 above)
Comment (deluxor) [reply db/models/users.js]: I just picked the schema from the git repo and placed it on the project

File: db/models/users.js (hunk 4 above)
Comment (deluxor) [reply db/models/users.js]: This seems to be a glitch on the generator from sequelize.

File: .github/workflows/nodejs.yml (hunk 1 above)
Comment (deluxor) [reply .github/workflows/nodejs.yml]: There are some specific features only supported on 13.x, for example async hooks, and they are really important in logging, which in this case i didn't have implemented yet.

Comment (hugoduraes) [reply .github/workflows/nodejs.yml]: Async hooks are still in experimental state for both 12.x and 13.x! Don't thin we should rely on them.

Comment (deluxor) [reply .github/workflows/nodejs.yml]: Well it was closed on https://github.com/nodejs/node/pull/31945 and released upstream in 13.10.0 https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V13.md#13.10.0
Need to wait for documentation to update.

**Pull request merged**